### DISCUSSION TITLE
Problem with variables: accessing variable in response prompt fails. Thanks for any help :)

### DISCUSSION THREAD

**Discussion by ori-aventure** [FlowiseAI/Flowise#1777] [Q&A] [answered]
Hi, thanks for taking the time to read this! I defined the static variable "tkp-name" as seen in the screenshot and I am trying to access it in my retrieval flow's response instructions but it fails. I don't know what to do because there is not much I could do differently as far as I can see. Any help is much appreciated!
![flow](https://github.com/FlowiseAI/Flowise/assets/152312095/e002306a-6673-4cc8-8474-73727ec3870a)
![variable-def](https://github.com/FlowiseAI/Flowise/assets/152312095/94941c45-5055-4ad0-abb3-8ce4831669be)
![prompt](https://github.com/FlowiseAI/Flowise/assets/152312095/ec1ffb55-0905-4264-be85-5453270df6f4)
![chat](https://github.com/FlowiseAI/Flowise/assets/152312095/f27d4cbe-3cc6-4a3c-a201-fad63c86f460)

**Comment by Jaredude** [comment, marked as answer]
Prompts don't have access to the variables like that. The variables are accessible within Custom Tools. You can read the docs on how to use variables here: https://docs.flowiseai.com/using-flowise/variables

**Reply by ori-aventure** [reply]
Thanks!!
